tempy/tempy.py

```diff
@@ -402,6 +406,51 @@ def wrap(self, other):
         other.append(self)
         return self
 
+    def wrap_many(self, *args, strict=False):
+        """Wraps different copies of this element inside all empty tags
+        listed in params or param's (non-empty) iterators.
+
+        Returns list of copies of this element wrapped inside args
+        or None if not succeeded, in the same order and same structure,
+        i.e. args = (Div(), (Div())) -> value = (A(...), (A(...)))
+
+        If on some args it must raise TagError, it will only if strict is True,
+        otherwise it will do nothing with them and return Nones on their positions"""
+        wcopies = []
+        failure = []
+
+        def wrap_next(tag, idx):
+            nonlocal wcopies, failure
+            next_copy = self.__copy__()
+            try:
+                return next_copy.wrap(tag)
+            except TagError:
+                failure.append(idx)
+                return None
+
+        arg_idx = -1
+        for arg in args:
+            arg_idx += 1
+            if isinstance(arg, DOMElement):
+                wcopies.append(wrap_next(arg, arg_idx))
+            else:
+                if not isinstance(arg, Iterable) or not arg or not isinstance(iter(arg).__next__(), DOMElement):
```

Comment:
I think this can be a little better, and can hide a little bug. The check is performed for each arg inside the loop, meaning that if the last arg is a bad arg the `WrongArgsError` is raised after doing all the copies and the wrapping of the previous args. Consider the following code:

```
good_image_badge = Img(src='img/good_badge.jpg')
images = [page.images.photos, page.images.drawings]
try:
    good_image_badge.wrap_many(images)
except:
    pass
```
If the last thing in the last arg contains a non DOMElement object the effect would be to add badges in all the previous images, but the desired behaviour is probably "try to add badges, if something goes wrong, no badges at all".

I'd move the `WrongArgsError` check at the beginning of the method, flattening the args list and checking for any non-DOMElement objects.